### ISSUE TITLE
Cannot backup after remote forget/prune on rest-server with --apend-only

### ISSUE THREAD

**Issue by deajan** [restic/restic#3963]
Output of `restic version`
--------------------------
```
restic 0.14.0 compiled with go1.19 on linux/amd64
```

How did you run restic exactly?
-------------------------------

I have a source server running restic and a target server running rest-server.
Since rest-server is run with `--append-only`, I need to forget/prune snapshots directly on the target server.

On the source server, I do a backup
```
export RESTIC_PASSWORD=SuperSecretPassword
export RESTIC_REPOSITORY='rest:https://someuser:somepassword@someserver.tld/someuser'
restic init --repository-version 2
restic backup --verbose --exclude-caches --tag="testa" -o rest.connections=25 --exclude ".git" --compression=max /some/path
```

On the target server, I delete that backup
```
export RESTIC_PASSWORD=SuperSecretPassword
export RESTIC_REPOSITORY='/local/path/to/repo/someuser'
# Let's forget the snapshot with tag testa I created above
restic forget $(restic snapshots | grep "testa" | awk '{print $1}')
# Now let's prune the repo
restic prune
```

Once this is done on target server, I try to run a second backup on source server:
```
 restic backup --verbose --exclude-caches --tag="testb" -o rest.connections=25 --exclude ".git" --compression=max /some/path
```

What backend/server/service did you use to store the repository?
----------------------------------------------------------------
```
rest-server version rest-server 0.11.0 compiled with go1.17.6 on linux/amd64
```
REST server is behind haproxy 2.4 which offloads TLS security.

rest server is run with following arguments:
```
/usr/local/bin/rest-server --path /stor/restic/ --listen :12345 --prometheus --prometheus-no-auth --append-only --private-repos
```
Expected behavior
-----------------

Source server does backup with tag "testa"
Removing snapshot with tag "testa" directly on target server.
source server does backup with tag "testb"

Actual behavior
---------------

Once I launch the second backup (with tag "testb") on source server,  I get the following output:
```
restic backup --verbose --exclude-caches --tag="testb" -o rest.connections=25 --exclude ".git" --compression=max /opt/backup_test
open repository
repository 9aa1b3dc opened (repository version 2) successfully, password is correct
lock repository
no parent snapshot found, will read all files
load index files
Load(<index/27c0f4e39a>, 0, 0) returned error, retrying after 552.330144ms: <index/27c0f4e39a> does not exist
Load(<index/27c0f4e39a>, 0, 0) returned error, retrying after 1.080381816s: <index/27c0f4e39a> does not exist
Load(<index/27c0f4e39a>, 0, 0) returned error, retrying after 1.31013006s: <index/27c0f4e39a> does not exist
Load(<index/27c0f4e39a>, 0, 0) returned error, retrying after 1.582392691s: <index/27c0f4e39a> does not exist
Load(<index/27c0f4e39a>, 0, 0) returned error, retrying after 2.340488664s: <index/27c0f4e39a> does not exist
Load(<index/27c0f4e39a>, 0, 0) returned error, retrying after 4.506218855s: <index/27c0f4e39a> does not exist
Load(<index/27c0f4e39a>, 0, 0) returned error, retrying after 3.221479586s: <index/27c0f4e39a> does not exist
Load(<index/27c0f4e39a>, 0, 0) returned error, retrying after 5.608623477s: <index/27c0f4e39a> does not exist
Load(<index/27c0f4e39a>, 0, 0) returned error, retrying after 7.649837917s: <index/27c0f4e39a> does not exist
Load(<index/27c0f4e39a>, 0, 0) returned error, retrying after 15.394871241s: <index/27c0f4e39a> does not exist
Load(<index/27c0f4e39a>, 0, 0) returned error, retrying after 507.606314ms: <index/27c0f4e39a> does not exist
Load(<index/27c0f4e39a>, 0, 0) returned error, retrying after 985.229971ms: <index/27c0f4e39a> does not exist
Load(<index/27c0f4e39a>, 0, 0) returned error, retrying after 803.546856ms: <index/27c0f4e39a> does not exist
Load(<index/27c0f4e39a>, 0, 0) returned error, retrying after 1.486109007s: <index/27c0f4e39a> does not exist
Load(<index/27c0f4e39a>, 0, 0) returned error, retrying after 2.070709754s: <index/27c0f4e39a> does not exist
Load(<index/27c0f4e39a>, 0, 0) returned error, retrying after 3.67875363s: <index/27c0f4e39a> does not exist
Load(<index/27c0f4e39a>, 0, 0) returned error, retrying after 4.459624189s: <index/27c0f4e39a> does not exist
Load(<index/27c0f4e39a>, 0, 0) returned error, retrying after 6.775444383s: <index/27c0f4e39a> does not exist
Load(<index/27c0f4e39a>, 0, 0) returned error, retrying after 15.10932531s: <index/27c0f4e39a> does not exist
Load(<index/27c0f4e39a>, 0, 0) returned error, retrying after 13.811796615s: <index/27c0f4e39a> does not exist
Fatal: <index/27c0f4e39a> does not exist
```

Steps to reproduce the behavior
-------------------------------
I hope you have everything required above ;)

Do you have any idea what may have caused this?
-----------------------------------------------
As far as I understand, using `--apend-only` on rest-server implies that the only place I can delete snapshots in the repository is directly on the server which runs rest-server.
I hope i didn't misunderstood anything here.

Do you have an idea how to solve the issue?
-------------------------------------------
???

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------

I'm the author of [backup-bench](https://github.com/deajan/backup-bench)
After a lot of benchmarking between all the solutions, I wanted to give restic a deeper test dive for it fills quite well my usecase.

**Comment by deajan** [comment]
Well... sorry for the noise.
I've run `restic prune` as root whereas the rest-server runs as `restic` user.
After a quick `chown -R restic:restic`, I'm good.

Inded the error message could be better, but I saw that was implemented in rest-server a couple of days ago.

I still have a side question, perhaps a bit out of scope.

I'm using `--append-only` so a compromised target cannot delete backups.
But in that case, I in order to make simple backup repo maintenance (eg deleting old snapshots, pruning...), I need to know the repo password, which allows me to access all data.
For my personal needs, that's just fine (I usually trust myself ;))
But if I'd use restic for customer backups, I would defacto be able to decrypt all their data without their knowings.

Anything I can do about this ?
The ideal solution IMO would be to have "operator_only" password which would allow to make basic housekeeping operations on the backup repo, but not decrypt / restore files.

**Comment by MichaelEischer** [comment]
Duplicate of #1871

> The ideal solution IMO would be to have "operator_only" password which would allow to make basic housekeeping operations on the backup repo, but not decrypt / restore files.

Right now, that is not possible, see #187. Even with asymmetric encryption, the user running prune would still be able to decrypt all directory metadata and likely fingerprint most files. Unless we totally change the repository format, to separate structure and other metadata. But the fingerprinting would still remain possible (maybe there is some fancy crypto to avoid that too, but I've no idea).
